File: dynamic_listings/python/default/developing_mechanics/coyote_time.py
class Player:
    coyote_time_started: bool = False
    coyote_time: float = 0
    has_jumped: bool = False
    is_on_ground:bool = False
    # ...

    def update(self, dt: float) -> None:
        # ...
        if self.is_on_ground:
            # Do stuff when player is on ground
            # ...
            pass
        else:
            if not self.has_jumped:
                # Player is not on ground and has not jumped, the player's falling
                if not self.coyote_time_started:
                    self.coyote_time_started = True
                    self.coyote_time = 5
                else:
                    self.coyote_time -= dt

    def jump(self) -> None:
        # This function takes care of jumping
        # ...
        if self.coyote_time > 0:
            # Do Jump
            pass
        # ...

File: dynamic_listings/rust/default/developing_mechanics/coyote_time.rs
# #![allow(dead_code)]
struct Player {
    coyote_time_started: bool,
    coyote_time: f32,
    onground: bool,
    has_jumped: bool,
}

impl Player {
    // ...
    fn update(&mut self, dt: f32) {
        // ...
        if self.onground {
            // Do stuff when player is on ground
            // ...
        } else {
            if !self.has_jumped {
                // Player is not on the ground and has not jumped, the player is falling
                if !self.coyote_time_started {
                    self.coyote_time_started = true;
                    self.coyote_time = 5.;
                } else {
                    self.coyote_time -= dt;
                }
            }
        }
    }

    fn jump(&mut self) {
        // This function takes care of jumping
        // ...
        if self.coyote_time > 0. {
            // Do Jump
            // ...
        }
    }
}
